File: src/main/java/org/cubexmc/features/revoke/RevokeConfirmation.java
package org.cubexmc.features.revoke;

import java.util.UUID;

public class RevokeConfirmation {
    private final String ruleKey;
    private final UUID targetUuid;
    private final String targetName;
    private final String powerKey;
    private final long expiresAtMillis;

    public RevokeConfirmation(String ruleKey, UUID targetUuid, String targetName, String powerKey,
            long expiresAtMillis) {
        this.ruleKey = ruleKey;
        this.targetUuid = targetUuid;
        this.targetName = targetName;
        this.powerKey = powerKey;
        this.expiresAtMillis = expiresAtMillis;
    }

    public String getRuleKey() { return ruleKey; }
    public UUID getTargetUuid() { return targetUuid; }
    public String getTargetName() { return targetName; }
    public String getPowerKey() { return powerKey; }
    public long getExpiresAtMillis() { return expiresAtMillis; }

    public boolean isExpired(long nowMillis) {
        return nowMillis > expiresAtMillis;
    }
}

File: src/main/java/org/cubexmc/features/revoke/RevokeConfirmation.kt
package org.cubexmc.features.revoke

import java.util.UUID

class RevokeConfirmation(
    val ruleKey: String,
    val targetUuid: UUID,
    val targetName: String,
    val powerKey: String,
    val expiresAtMillis: Long,
) {
    fun isExpired(nowMillis: Long): Boolean = nowMillis > expiresAtMillis
}
